File: Data Structures & Algorithms/decode-ways/submission-21.cpp
class Solution {
public:
   
    int numDecodings(string s) {

        int n = s.size();

        return dfs(0,s);   
    }
private:
   int dfs(int i, string& s){
    if(s[i]=='0') return 0;
    if(i==s.size()) return 1;

    int res = dfs(i+1,s);
    if(i<s.size()-1){
      if(s[i]=='1' || s[i] == '2' && s[i+1]<'7'){
        res+=dfs(i+2,s);
      }
    }
    return res;


   }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-21.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = defaultdict(int)
        l, res = 0,0

        for r in range(len(s)):
            count[s[r]]+=1

            while count[s[r]]>1:
                count[s[l]]-=1
                l+=1
            
            res = max(res,r-l+1)
        
        return res


        